srcts/src/shiny/modal.ts

```diff
@@ -72,7 +88,14 @@ function remove(): void {
   // trigger the hidden.bs.modal callback that we set in show(), which unbinds
   // and removes the element.
   if ($modal.find(".modal").length > 0) {
-    $modal.find(".modal").modal("hide");
+    const $bsModal = $modal.find(".modal");
+    if (modalsInTransition.has($bsModal[0])) {
+      // If the modal is transitioning, we need to schedule the hide event to
+      // happen once the modal is fully revealed.
+      $bsModal.on("shown.bs.modal", () => $bsModal.modal("hide"));
+    } else {
+      $bsModal.modal("hide");
```

Comment:
Is it possible to call both at all times?

Then we could remove the WeakSet and other `on` methods added earlier.

If the modal is transitioning, then `.modal("hide")` would be a no-op and the `show.bs.modal` will hide it. 
If the modal is displayed, then adding the event listener would be a minimal cost compared to the logic cost of the WeakSet.